java-api-core/src/com/kinvey/java/network/NetworkManager.java

```diff
@@ -468,6 +462,18 @@ public String onUnsavedReferenceFound(String collection, GenericJson object) {
         return save;
     }
 
+    public boolean isTempId(T item) {
+        boolean isTempId = true;
+        if (item.get(Constants._ID) != null) {
```

Comment:
Why is the default value here `true`?  Would it be better if it were `false`, so we only return `true` if we are certain that the format of the `_id` is the temporary format?

Reply:
You are right. And if the SDK gets `true` from this method then the POST request will be made and if it was not temp id then we will have duplicate items at the backend with different ids. Updated.